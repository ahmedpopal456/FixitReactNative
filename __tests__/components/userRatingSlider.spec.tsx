import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UserRatingSlider from '../../src/components/userRatingSlider';

describe('User Rating Slider', () => {
  const props = {
    score: 3,
  };

  it('renders correctly', async () => {
    const userRatingSlider = renderer.create(
      <UserRatingSlider {...props} />,
    ).toJSON();
    expect(userRatingSlider).toMatchSnapshot();
  });

  it('should display proper caret color', () => {
    const wrapper = shallow(
      <UserRatingSlider {...props} />,
    );
    expect(wrapper.instance().calculateCaretColor(1)).toEqual('#D4675A');
  });

  it('should calculate the caret offset', () => {
    const wrapper = shallow(
      <UserRatingSlider {...props} />,
    );
    expect(wrapper.instance().calculateCaretOffset()).toEqual(4.05);
  });
});
