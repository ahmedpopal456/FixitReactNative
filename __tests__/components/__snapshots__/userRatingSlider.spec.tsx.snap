// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Rating Slider renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 20,
      "marginTop": -15,
    }
  }
>
  <View>
    <Text
      allowFontScaling={false}
      library="FontAwesome5"
      style={
        Array [
          Object {
            "color": "#33343E",
            "fontSize": 15,
          },
          Object {
            "color": "#D4675A",
            "marginLeft": 4.05,
          },
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      ïƒ—
    </Text>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D4675A",
            "borderRadius": 2,
            "height": 4,
            "width": 45,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#E58346",
            "borderRadius": 2,
            "height": 4,
            "marginLeft": -2,
            "width": 30,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#F5D23E",
            "borderRadius": 2,
            "height": 4,
            "marginLeft": -2,
            "width": 25,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#5690FA",
            "borderRadius": 2,
            "height": 4,
            "marginLeft": -2,
            "width": 25,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#8ED698",
            "borderRadius": 2,
            "height": 4,
            "marginLeft": -2,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "marginLeft": 10,
        "marginTop": 8,
      }
    }
  >
    3
  </Text>
</View>
`;
